dora/core/common/src/main/java/alluxio/uri/UfsUrl.java

```diff
@@ -0,0 +1,486 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.uri;
+
+import alluxio.AlluxioURI;
+import alluxio.conf.Configuration;
+import alluxio.conf.PropertyKey;
+import alluxio.exception.InvalidPathException;
+import alluxio.grpc.UfsUrlMessage;
+import alluxio.util.io.PathUtils;
+
+import com.google.common.base.Preconditions;
+import org.apache.commons.io.FilenameUtils;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * This class represents a UFS URL in the Alluxio system. This {@link UfsUrl} represents the
+ * absolute ufs path.
+ */
+public class UfsUrl {
+
+  public static final String SCHEME_SEPARATOR = "://";
+  public static final String DOUBLE_SLASH_SEPARATOR = "//";
+  public static final String PATH_SEPARATOR = "/";
+  public static final String PORT_SEPARATOR = ":";
+
+  final UfsUrlMessage mProto;
+
+  /**
+   * Constructs a map containing scheme, authority, and path.
+   * @param inputUrl the input url string
+   * @return a map containing scheme, authority, and path
+   */
+  public static Map<String, String> extractElements(String inputUrl) {
+    Preconditions.checkArgument(inputUrl != null && !inputUrl.isEmpty(),
+        "The input url is null or empty, please input a valid url.");
+    Map<String, String> elements = new HashMap<>();
+    String scheme = null;
+    String authority = null;
+    String path = null;
+
+    int start = 0;
+    int schemeSplitIndex = inputUrl.indexOf(SCHEME_SEPARATOR, start);
+    if (schemeSplitIndex == -1) {
+      scheme = "";
+    } else {
+      scheme = inputUrl.substring(start, schemeSplitIndex);
+      start += scheme.length() + SCHEME_SEPARATOR.length();
+    }
+    Preconditions.checkArgument(!scheme.equalsIgnoreCase("alluxio"),
+        "Alluxio 3.x no longer supports alluxio:// scheme,"
+            + " please input the UFS path directly like hdfs://host:port/path");
```

Comment:
```suggestion
    Preconditions.checkArgument(!scheme.equalsIgnoreCase(Constants.SCHEME),
        "Expected a UFS URL, but found Alluxio URL: %s", inputUrl);
```

Reply:
I think it makes sense to mention that `alluxio://` scheme is no longer supported. Users come with "alluxio://" are from 2.x and will get confused if they see "Expect a UFS URL". I think we should combine your comment with mine (the existing error msg).